libs/server/Resp/Objects/HashCommands.cs

```diff
@@ -31,66 +31,65 @@ private unsafe bool HashSet<TGarnetApi>(RespCommand command, int count, byte* pt
             {
                 return AbortWithWrongNumberOfArguments(command.ToString(), count);
             }
-            else
-            {
-                // Get the key for Hash
-                if (!RespReadUtils.ReadByteArrayWithLengthHeader(out var key, ref ptr, recvBufferPtr + bytesRead))
-                    return false;
 
-                if (NetworkSingleKeySlotVerify(key, false))
-                {
-                    return true;
-                }
+            var sbKey = parseState.GetArgSliceByRef(0).SpanByte;
+            var keyBytes = sbKey.ToByteArray();
 
-                // Prepare input
-                var inputPtr = (ObjectInputHeader*)(ptr - sizeof(ObjectInputHeader));
+            var ptr = sbKey.ToPointer() + sbKey.Length + 2;
```

Comment:
Nit: A lot of places repeat this pattern ToPointer + Length + 2. Could be extracted into a small inline/extension method.